src/arrayMethodJoin.js

```diff
@@ -4,9 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === undefined || this[i] === null) {
+        this[i] = '';
+      }
+
+      if (result.length === 0 && this.length !== 1) {
+        result = `${this[i]}${separator}`;
+      } else if (i + 1 !== this.length) {
+        result = `${result}${this[i]}${separator}`;
+      } else {
```

Comment:
avoid` if else `construction, two if's are enough for you to make the necessary checks